src/app/components/SearchBox/SearchBox.tsx

```diff
@@ -1,25 +1,45 @@
 'use client';
 
-import React, { FormEvent, useState } from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
 import { useRouter } from 'next/navigation';
 import Image from 'next/image';
+import useDebounce from '@/hooks/useDebounce';
+import { useLoadingContext } from '@/context/loading.context';
 import styles from './SearchBox.module.scss';
 
 function SearchBox() {
   const [input, setInput] = useState('');
+  const { loading } = useLoadingContext();
   const router = useRouter();
+  const debouncedInput = useDebounce(input, 1000);
 
-  const handleSearch = (e: FormEvent<HTMLFormElement>) => {
-    e.preventDefault();
-    router.push(`?q=${input}`);
-  };
+  const handleSearch = useCallback(() => {
+    if (debouncedInput) {
+      router.push(`?q=${debouncedInput}`);
+    }
+  }, [debouncedInput, router]);
+
+  useEffect(() => {
+    handleSearch();
+  }, [debouncedInput, handleSearch]);
 
   return (
-    <form className={styles.searchBox} onSubmit={handleSearch}>
-      <input type="text" placeholder="Search keywords..." value={input} onChange={(e) => setInput(e.target.value)} />
-      <button type="submit" disabled={!input}>
+    <form
+      className={styles.searchBox}
+      onSubmit={(e) => {
+        e.preventDefault();
+      }}
+    >
+      <input
+        type="text"
+        placeholder="Search keywords..."
+        value={input}
+        onChange={(e) => setInput(e.target.value)}
+        disabled={loading}
+      />
+      <span className={!input ? styles.searchBox__disabled : ''}>
```

Comment:
If you want to use conditions for classes then install any lib for that to make it more readable